packages/server-render/server-register.js

```diff
@@ -1,10 +1,58 @@
 import { WebAppInternals } from "meteor/webapp";
+import cheerio from "cheerio";
 import MagicString from "magic-string";
 import { SAXParser } from "parse5";
 import { create as createStream } from "combined-stream2";
 import { ServerSink, isReadable } from "./server-sink.js";
 import { onPageLoad } from "./server.js";
 
+function updateContent(data, section, selector, itemArray) {
+  let reallyMadeChanges = false;
+  const content = section === "head" ?
+    `<head>${data.head}</head>` :
+    `<body>${data.body}</body>`;
+  $ = cheerio.load(content,
+    { 
+      withDomLvl1: true,
+      normalizeWhitespace: false,
+      xmlMode: true,
+      decodeEntities: true
+    }
+  );
+
+  itemArray.forEach((item) => {
+    let searchField = "";
+    if (selector === "id") {
+      searchField = `#${item.id}`;
+    } else {
+      searchField = item.searchAttribute && item.searchAttributeValue ?
+        `${item.tag}[${item.searchAttribute}=${item.searchAttributeValue}]` :
+        item.tag;
+    }
+    if (!item.value) {
+      $(searchField).remove();
+    } else {
+      if (item.updateAttribute) {
+        $(searchField).attr(item.updateAttribute, item.value);
+      } else {
+        if (item.updateType === 'prepend') {
+          $(searchField).prepend(item.value);
+        } else if (item.updateType === 'append') {
+          $(searchField).append(item.value);
+        } else {
+          $(searchField).html(item.value);
+        }
+      }
+      reallyMadeChanges = true;
+    }
+  });
+  if (reallyMadeChanges) {
+    data[section] = $(section).html();
+  }
+
+  return reallyMadeChanges;
+}
```

Comment:
Please use (or improve) the existing `rewrite` pattern, based on the `parse5` streaming `SAXParser`, rather than introducing a new parsing technique. Specifically, the pattern you're using here doesn't allow for a streaming response, because it takes whole chunks of HTML and replaces them after all transforms have completed. Especially since you're modifying the `<head>`, this pattern has the potential to delay rendering of the rest of the page, since the `<head>` has to be rendered first.